lab-david/__test__/photo-router.test.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+const request = require('superagent');
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('fs'), { suffix: 'Prom'});
+const debug = require('debug')('cfgram:photo-router.test');
+const server = require('../server.js');
+const serverToggle = require('../lib/server-toggle.js');
+
+const Photo = require('../model/photo.js');
+const User = require('../model/user.js');
+const PhotoAlbum = require('../model/photoalbum.js');
+
+require('jest');
+
+const url = 'http://localhost:3000';
+
+const exampleUser = {
+  username: 'exampleuser',
+  password: '1234',
+  email: 'exampleuser@user.net',
+};
+
+const examplePhotoAlbum = {
+  name: 'example album',
+  desc: 'example album desc',
+};
+
+const examplePhoto = {
+  name: 'example photo',
+  desc: 'example photo desc',
+  image: `${__dirname}/../data/tester.png`,
+};
+
+describe('Photo Router', function(){
+  beforeAll( done => {
+    serverToggle.serverOn(server, done);
+  });
+
+  afterAll( done => {
+    serverToggle.serverOff(server, done);
+  });
+
+  beforeAll( done => {
+    new User(exampleUser)
+      .generatePasswordHash(exampleUser.password)
+      .then( user => user.save())
+      .then( user => {
+        this.tempUser = user;
+        return user.generateToken();
+      }).then( token => {
+        this.tempToken = token;
+        done();
+      }).catch(done);
+  });
+
+  beforeAll( done => {
+    examplePhotoAlbum.userId = this.tempUser._id.toString();
+    new PhotoAlbum(examplePhotoAlbum).save()
+      .then( album => {
+        this.tempAlbum = album;
+        done();
+      });
+  });
+
+  this.tempPhoto = {};
+
+  afterAll( done => {
+    Promise.all([
+      Photo.remove({}),
+      User.remove({}),
+      PhotoAlbum.remove({}),
+    ])
+      .then( () => done())
+      .catch(done);
+  });
+
+  describe('POST: /api/photoalbum/:photoalbumId/photo', () => {
+    describe('with a valid token and valid data', () => {
+
+      beforeEach( done => {
```

Comment:
Interesting approach. Way to get around the issue of the photo getting removed. Kinda strange to just have the tester.png floating in your root. Since it is only needed for testing, I would create a data folder in your __test__ directory and put it there. This way you can read that file in, make the multer copy to the other directory, and then remove it without losing the original.